src/v/storage/CMakeLists.txt

```diff
@@ -47,6 +47,7 @@ v_cc_library(
     v::compression
     v::rprandom
     v::resource_mgmt
+    v::cluster
```

Comment:
Don't think we should make storage depend on cluster, that creates a circular dependency.

I believe you added it to pass _topic_table into storage. To avoid this, how about making the function definition into

```
ss::future<> log_manager::remove_orphan_files(
  ss::sstring data_directory_path,
  absl::flat_hash_set<model::ns> namespaces,
  ss::noncopyable_function<bool(model::ntp, partition_path::metadata)> orphan_filter
 )
``` 

and then pass the filter as lambda from controller backend that captures all the needed context, something like.. (need to move some code around)

```
ss::future<> controller_backend::clear_orphan_topic_files(
  model::revision_id bootstrap_revision) {
    // Init with default namespace to clean if there is no topics
    absl::flat_hash_set<model::ns> namespaces = {{model::kafka_namespace}};
    for (const auto& t : _topics.local().all_topics()) {
        namespaces.emplace(t.ns);
    }

    return _storage.local().log_mgr().remove_orphan_files(
      _data_directory,
      std::move(namespaces),
      [&, bootstrap_revision](
        model::ntp ntp, storage::partition_path::metadata p) {
          return topic_files_are_orphan(
            ntp, p, _topics, bootstrap_revision, _self);
      });
}
```

Reply:
Great idea!
Thank you